### DISCUSSION TITLE
[Feature Request] Implement IL Code viewer

### DISCUSSION THREAD

**Discussion by realvictorprm** [ionide/ionide-vscode-fsharp#418] [Ideas]
Would be nice to have an IL Code viewer in Ionide too. 
_The idea came up by a person on the FSSF slack channel called @b0urb4k1._

**Comment by Krzysztof-Cieslak** [comment]
In principle sounds good. But I have no idea how to get IL for the assembly etc.

**Comment by jakzale** [comment]
There is a .net disassembler provided with coreclr,  [ildasm](https://docs.microsoft.com/en-us/dotnet/framework/tools/ildasm-exe-il-disassembler), which I guess the .net core community is planning to release as a separate tool at some point (dotnet/cli#6223).

Currently, `ildasm` is available as a [nuget package](https://www.nuget.org/packages/Microsoft.NETCore.ILDAsm/).  I played around` ildasm` and I am writing a small proof-of-concept of using it to show IL for an assembly.

**Comment by 7sharp9** [comment]
IL code viewer for referenced assemblies, or for in project files?

**Comment by Krzysztof-Cieslak** [comment]
Preferably both?

**Comment by a-sayapin** [comment]
there definetely is an example of disassembling IL in "C# in a nutshell" (if I'm not mistaken)
BTW, how should that look like? like preview window for markdown?
@Krzysztof-Cieslak, @jakzale if you're ok with this, can I pick it?
